presto-common/src/main/java/com/facebook/presto/common/block/ArrayBlock.java

```diff
@@ -36,27 +36,29 @@
     private final int[] offsets;
 
     private volatile long sizeInBytes;
+    private volatile long logicalSizeInBytes;
     private final long retainedSizeInBytes;
 
     /**
      * Create an array block directly from columnar nulls, values, and offsets into the values.
      * A null array must have no entries.
      */
-    public static Block fromElementBlock(int positionCount, Optional<boolean[]> valueIsNull, int[] arrayOffset, Block values)
+    public static Block fromElementBlock(int positionCount, Optional<boolean[]> valueIsNullOptional, int[] arrayOffset, Block values)
     {
-        validateConstructorArguments(0, positionCount, valueIsNull.orElse(null), arrayOffset, values);
+        boolean[] valueIsNull = valueIsNullOptional.orElse(null);
+        validateConstructorArguments(0, positionCount, valueIsNull, arrayOffset, values);
         // for performance reasons per element checks are only performed on the public construction
         for (int i = 0; i < positionCount; i++) {
             int offset = arrayOffset[i];
             int length = arrayOffset[i + 1] - offset;
             if (length < 0) {
                 throw new IllegalArgumentException(format("Offset is not monotonically ascending. offsets[%s]=%s, offsets[%s]=%s", i, arrayOffset[i], i + 1, arrayOffset[i + 1]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;arrayget()&#39; expression, &#39;+&#39; expression, i (declared at line 51). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 55. Consider formatting this data with &#39;%d&#39; instead.